### PR TITLE
Implement events

### PR DESCRIPTION
**Pull request by edahigure**
- Create a new directory called logic inside your project
- Copy the files calculate.js and operate.js into your logic/ directory
- Install [big.js](https://www.npmjs.com/package/big-js) using NPM
- Analyze the files calculate.js and operate.js, be sure you understand what they do
- Import the files in your Calculator.js component
- Implement click event handlers using the math logic from calculate.js and operate.js in your React component
- Test your app in the browser. The Calculator should be able to perform math operations.

### CODE REVIEW

**Review by Danieldotcomcoder — COMMENTED** [review]
Hi @edahigure , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

1) Kindly make sure to remove the template logo from your Readme file,  you can replace with your own logo if you have one.

2) Kindly make sure to remove the section description in your Readme file, i.e. ">Describe 1 - 3 features you will add to the project" should be removed

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,44 +1,78 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
 // eslint-disable import/no-extraneous-dependencies
 
 // eslint-disable-next-line react/prefer-stateless-function
 class Calculator extends React.Component {
   // eslint-disable-next-line no-useless-constructor
```
Comment (mikethreels) [inline src/components/Calculator.js]: - [ ] please avoid turning off linters they are used to improve your code and make it more readable.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,44 +1,78 @@
...
+    const { total, next, operation } = this.state;
+    const obj = calculate(({ total, next, operation }), buttonName);
+
+    this.setState(() => ({
+      total: obj.total,
+      next: obj.next,
+      operation: obj.operation,
+    }));
+    console.log('buttonName', buttonName);
+    console.log('obj', obj);
```
Comment (mikethreels) [inline src/components/Calculator.js]: - [ ] please don't push any console.log to the repo they will make your code look unorganised and harder to read.

File: src/components/Calculator.js [hunk 3]
```diff
@@ -1,44 +1,78 @@
...
 
+  handleClick = (e) => {
+    const buttonName = e.target.textContent;
+    const { total, next, operation } = this.state;
+    const obj = calculate(({ total, next, operation }), buttonName);
+
+    this.setState(() => ({
+      total: obj.total,
+      next: obj.next,
+      operation: obj.operation,
```
Comment (mikethreels) [inline src/components/Calculator.js]: - [ ] please handle your state using the useState hook.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Edahi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 3 above)
Comment (edahigure) [reply src/components/Calculator.js]: Hi Mike, this project for yesterday requirements, so it is not supposed to use useState for it. 
Notice here [yesterday project](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md).
Yesterday project requirements:

Create a new directory called logic inside your project
Copy the files calculate.js and operate.js into your logic/ directory
Install [big.js](https://www.npmjs.com/package/big-js) using NPM
Analyze the files calculate.js and operate.js, be sure you understand what they do
Import the files in your Calculator.js component
Implement click event handlers using the math logic from calculate.js and operate.js in your React component
Test your app in the browser. The Calculator should be able to perform math operations.
Note: Practice what you have learned until now: use useState and props

Please review my project again so as to go to next step. Thank you @mikethreels

File: src/components/Calculator.js [hunk 4]
```diff
@@ -1,44 +1,73 @@
...
+    }));
```
Comment (rustam-tolipov) [inline src/components/Calculator.js]: - [ ] As the previous reviewer requested, kindly add useState to your app. You need to practice what you have learned about state/useState. The project [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md#:~:text=Note%3A%20Practice%20what%20you%20have%20learned%20until%20now%3A%20use%20useState%20and%20props.) that you submitted.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @edahigure, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
_Well documented Readme_

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
## Hi 🙋 @edahigure  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights </img>

- Calculator has been refactored to use useState hooks 👍 
- Linters checks passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**